api/api.go

```diff
@@ -42,10 +42,11 @@ type TaskEnd struct {
 }
 
 type SearchRequest struct {
-	Space SpaceID         `json:"space" validate:"required"`
-	Proc  json.RawMessage `json:"proc,omitempty"`
-	Span  nano.Span       `json:"span"`
-	Dir   int             `json:"dir" validate:"required"`
+	Space   SpaceID         `json:"space" validate:"required"`
+	Proc    json.RawMessage `json:"proc,omitempty"`
+	Span    nano.Span       `json:"span"`
+	Dir     int             `json:"dir" validate:"required"`
+	Workers int             `json:"workers"`
```

Comment:
I agree that it shouldn't be part of the public search api. 

@marktwallace : for this PR, I suggest you define a const "maxWorkersPerRequest" somewhere with a largish number, and have the root send that to the recruiter? In the future, we'll work on limiting the number requested based on analyzing the query & the data set.

Reply:
I will add my reason for including it: ultimately, the resources committed to a query must be based on the end user's intent. At some point, the user must specify their desired resource commitment, e.g. signing up for a plan;  specifying preferences for a session. Information about desired parallelism will vary based on the origin of the query, so should be carried along with it. 
Queries to the same cluster will have reasons to have different levels of resource commitment.

Reply:
This is not intended to be part of the public API. My plan is that the API server will gather information about desired threads from something about the users profile (e.g. whether they use a free or a premium plan). Then the API server will pass that information to the zqd root process, which will use it to request workers from the                      recruiter. This actually need to be a request parameter, not a global. It just looks confusing now because the API server has not been developed and deployed yet. 

It might clarify the issue if I introduce a "/root/request" route in core and only allow the "Workers" parameter for that request.

Reply:
I removed Workers from the public search API. The approach I took was to more fully develop the /worker API. Now there are two types of /worker requests:
/worker/chunksearch  is like the old /worker request (it includes chunk details)
/worker/rootsearch will be the internal call from the API server to the root worker. It includes number_to_recruit.
I expect more information will be added to /worker/rootsearch as we build out the cloud product experience.